sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/AzureConfiguration.java

```diff
@@ -3,13 +3,59 @@
 
 package com.azure.data.appconfiguration;
 
+import com.azure.core.util.logging.ClientLogger;
+import java.io.InputStream;
+import java.util.Properties;
+
 /**
  * Gets the SDK information for this library component.
  */
 class AzureConfiguration {
-    //TODO: Eventually remove these hardcoded strings with https://github.com/Azure/azure-sdk-for-java/issues/3141
-    static final String NAME = "application-configuration";
-    // {x-version-update-start;com.azure:azure-data-appconfiguration;current}
-    static final String VERSION = "1.0.0-preview.6";
-    // {x-version-update-end}
+
+    private static final ClientLogger LOGGER = new ClientLogger(AzureConfiguration.class);
+    private static final String NAME;
+    private static final String VERSION;
+    private static final String VERSION_NOT_DEFINED = "Version not defined";
+    private static final String ARTIFACTID_NOT_DEFINED = "ArtifactId not defined";
+    private static final String AZURE_APPCONFIG_PROPERTIES = "azure-appconfig.properties";
+    private static final String VERSION_PROPERTY_NAME = "version";
+    private static final String ARTIFACT_PROPERTY_NAME = "artifact";
+
+    static {
+        String version = null;
```

Comment:
Do we need to define these as null? Instead could we set them to the default values you have.

Reply:
The `getProperty()` method below might return `null` if the property is not set. So, the last 2 lines that do the null check and set to default is still required. So, I didn't set it to default here.